inst/scripts/2.4.0/make-hermann-spermatogenesis-data.Rmd

```diff
@@ -0,0 +1,205 @@
+---
+title: Processing the Hermann spermatogenesis dataset for RNA velocity analysis
+author: Charlotte Soneson
+date: "June 5, 2020"
+output:
+  BiocStyle::html_document:
+    titlecaps: false
+    toc_float: true
+bibliography: ../ref.bib
+---
+
+```{r style, echo=FALSE, results='hide', message=FALSE}
+library(BiocStyle)
+knitr::opts_chunk$set(error=FALSE, message=FALSE, warning=FALSE, eval=FALSE)
+```
+
+# Loading required R packages
+
+```{r}
+suppressPackageStartupMessages({
+    library(Biostrings)
+    library(BSgenome)
+    library(eisaR)
+    library(GenomicFeatures)
+    library(SummarizedExperiment)
+    library(tximeta)
+    library(rjson)
+    library(SingleCellExperiment)
+    library(dplyr)
+})
+```
+
+# Setting the working directory
+
+```{r}
+workdir <- tempdir()
+Sys.setenv(workdir = workdir)
+```
+
+# Downloading the reference sequences
+
+In order to perform the quantification of spliced and unspliced abundances with alevin [@srivastava2019alevin], we need to first prepare a suitable set of reference features (transcript isoform and intron sequences) to generate an index from. 
+We start by downloading the mouse reference genome and annotation gtf file from GENCODE.
+
+```{bash}
```

Comment:
Consider using **BiocFileCache** to mediate the download so that it is only run once.

Reply:
I could do this for the GTF, but I'm not sure it would make things easier in general, since I'm using the genome also in the bash chunks (and additionally gunzip it before, so I guess I'd have to keep both the compressed and the uncompressed files). The download time for the annotation files is a tiny fraction of the total processing time.